.github/CONTRIBUTING.md

```diff
@@ -22,6 +26,23 @@ The long version:
 
 *Please* provide feedback about how these practices impact your ability to contribute. You might voice that feedback on the [Apereo licensing discussion Google Group][] or in any other way in which you are comfortable.
 
+## Conventional Commits
+
+This project uses [Conventional Commits][].
+
+Conventional Commits compliance is not a hard requirement to contribute. Do
+what you can. If necessary commit message style can be further worked up at the
+Pull Request / code review layer. Non-compliant commit messages will be flagged
+by continuous integration (e.g. Travis-CI) on pull requests.
+
+You can locally lint (check for style compliance) your most recent commit message by `npm run lint-commit`.
+
+You can locally lint your commit messages via a pre-commit hook by
+
+1. Setting `"commitmsg": "commitlint -e",` in your local `package.json`
+2. Having a `.git/commit-msg` script that honors the `package.json`
```

Comment:
This could work, I added some notes at https://github.com/uPortal-Project/uportal-app-framework/issues/535#issuecomment-348574614 on how this could be normalized to remove some of the manual effort.